unit/adsr.go

```diff
@@ -0,0 +1,144 @@
+package unit
+
+import (
+	"math"
+
+	"buddin.us/shaden/dsp"
+)
+
+func newAdsr(name string, _ Config) (*Unit, error) {
+	io := NewIO()
+	return NewUnit(io, name, &adsr{
+		state:       &adsrState{},
+		stateFunc:   adsrIdle,
+		gate:        io.NewIn("gate", dsp.Float64(0)),
+		attack:      io.NewIn("attack", dsp.Duration(50)),
+		decay:       io.NewIn("decay", dsp.Duration(50)),
+		sustain:     io.NewIn("sustain", dsp.Float64(0.5)),
+		sustainHold: io.NewIn("sustain-hold", dsp.Duration(50)),
```

Comment:
One more question: What about defaulting this to zero? I see you have it skipping the sustain stage when it is zero. Thoughts?